crates/subspace-networking/src/node_runner.rs

```diff
@@ -509,6 +509,24 @@ where
                 info.listen_addrs.truncate(30);
             }
 
+            let addresses = info
+                .listen_addrs
+                .iter()
+                .filter(|addr| {
+                    // filter non-global addresses when non-globals addresses are disabled
+                    is_global_address_or_dns(addr) || self.allow_non_global_addresses_in_dht
+                })
+                .cloned()
+                .collect::<HashSet<_>>() // deduplication
+                .into_iter()
+                .collect::<Vec<_>>();
+
+            if !addresses.is_empty() {
+                self.networking_parameters_registry
+                    .add_known_peer(peer_id, addresses)
+                    .await;
+            }
```

Comment:
I recall previously we were also checking if it is an address presented by peer (and thus not necessarily dial-able) or is it an address we have just dialed successfully. We wanted to only store those that we are sure work unless I forgot something. Still no idea how we lost that code in the process.